lib/requester/request-wrapper.js

```diff
@@ -78,16 +79,18 @@ var _ = require('lodash'),
     setProxy = function(options, cb) {
         var url = _.get(options, 'request.url');
 
-        if (url && options.proxyManager && options.proxyManager.getProxyConfiguration) {
+        if (url && options.proxyList) {
+            if(!_.isFunction(options.proxyList.isProxyConfigList)) {
+              options.proxyList = new ProxyConfigList({}, options.proxyList);
```

Comment:
Runtime assumes that the incoming parameters are sanitised. Can we not assume that incoming proxyList option if truthy, is a proxy list and proceed without cloning it?

That would also mean no dependency on collection SDK. Also it would not clone the list and use existing object. It is the consumer of runtime who decides what to give cloned and runtime can mutate the options - like it does for variable scope.